frontend/packages/ceph-storage-plugin/src/components/dashboards/odf-system-dashboard.tsx

```diff
@@ -31,23 +33,26 @@ const ODFSystemDashboard: React.FC<ODFSystemDashboardPageProps> = ({
   const { systemName } = rest.match.params;
   const dashboardTab = !isCephAvailable && isObjectServiceAvailable ? OBJECT : BLOCK_FILE;
   const defaultDashboard = React.useRef(dashboardTab);
-
-  const pages: Page[] = [
+  const [pages, setPages] = React.useState<Page[]>([
     {
       path: 'overview/:dashboard',
       href: `overview/${defaultDashboard.current}`,
       name: t('ceph-storage-plugin~Overview'),
       component: OCSOverview,
     },
-  ];
+  ]);
 
   React.useEffect(() => {
-    if (isCephAvailable) {
-      pages.push({
-        href: referenceForModel(CephBlockPoolModel),
-        name: t('ceph-storage-plugin~BlockPools'),
-        component: () => <BlockPoolListPage namespace={CEPH_STORAGE_NAMESPACE} />,
-      });
+    const isBlockPoolAdded = pages.find((page) => page.href === blockPoolRef);
+    if (isCephAvailable && !isBlockPoolAdded) {
+      setPages([
+        ...pages,
+        {
+          href: blockPoolRef,
+          name: t('ceph-storage-plugin~BlockPools'),
+          component: () => <BlockPoolListPage namespace={CEPH_STORAGE_NAMESPACE} />,
+        },
+      ]);
```

Comment:
```
const pages: Page[] = [
    {
      path: 'overview/:dashboard',
      href: `overview/${defaultDashboard.current}`,
      name: t('ceph-storage-plugin~Overview'),
      component: OCSOverview,
    },
    ... isCephAvailable ?  [{
      href: referenceForModel(CephBlockPoolModel),
      name: t('ceph-storage-plugin~BlockPools'),
      component: () => <BlockPoolListPage namespace={CEPH_STORAGE_NAMESPACE} />,
    }] : []
  ];
```

Reply:
This wouldn't work. The reason it was previously not working is React rerenders only when state or props change. Since `const pages` is neither. This will not rerender. `isCephAvailable` is resolved after some time by which time the component is already rendered so this isn't much different from the original code.

Reply:
ack, is there any problem of using let instead of const, just to know?